### PR TITLE
feat: add and setup C3 configuration provider component to operate

### PR DESCRIPTION
**Pull request by vitorwtavares**
## Description
- Adds and configures C3UserConfigurationProvider to Operate
- Refactor of App Header
- Addition of teaser pages for other apps

## Related issues

closes camunda/camunda#32831

### CODE REVIEW

File: operate/client/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "@bpmn-io/element-template-icon-renderer": "0.5.2",
     "@camunda/camunda-composite-components": "0.7.0",
     "@carbon/elements": "11.46.0",
-    "@carbon/react": "1.58.1",
+    "@carbon/react": "1.57.0",
```
Comment (pedesen) [inline operate/client/package.json:10]: This is 1.57.0 on `main` now. So you should be able to rebase. Then there should be no changes needed for package.json and yarn.lock

**Review by pedesen — CHANGES_REQUESTED** [review]

**Review by pedesen — APPROVED** [review]
The visual regression snapshots are part of the changelog even though they are identical. I think you could squash the two commits https://github.com/camunda/camunda/pull/18189/commits/2d84c69de4ad493fc6f12b0bb67cb9785dea595e and https://github.com/camunda/camunda/pull/18189/commits/3d542cf7761919946ffc8c7d608a2559c12ed2e8 to avoid this. But it's not a big deal.

**Comment by Szik** [comment]
Bildschirmfoto vom 2024-05-29 19-34-32

List of Camunda Components is empty

**Review by Szik — CHANGES_REQUESTED** [review]
Camunda Compontents list is empty

**Comment by vitorwtavares** [comment]
Closing this PR as it's stale and this feature is not high priority at the moment. The branch will be kept so we can update it and keep the work already done whenever we decide to pick this back up.

**Pull request closed without merging**